python/tvm/tir/schedule/_type_checker.py

```diff
@@ -134,14 +169,25 @@ def _type_check_atomic(v: Any, name: str, type_: Any) -> Optional[str]:
         return None if isinstance(v, type_) else _type_check_err(v, name, type_)
 
     def _type_check_list(v: List[Any], name: str, type_: Any) -> Optional[str]:
-        if not isinstance(v, (list, tuple)):
+        if not isinstance(v, list):
```

Comment:
implicit cast from tuple to list is allowed if the element types are the same

Reply:
Got it.  I wasn't sure if that was intended behavior, or was an early version of supporting tuples.  It makes sense for it to be intended behavior, since both are converted to `Array<T>` by the ffi.  I will revert this line.